### ISSUE TITLE
Deployment of a new daemonset cause CA "scale-from-zero" behavior unstable.

### ISSUE THREAD

**Issue by xinxingliu90** [kubernetes/autoscaler#2927]
## Problem
Some ASGs are not able to scale from zero. Pods pending forever.

## Change
A device-plugin is deployed to cluster as K8s daemonset. Therefore for each pod, a new request is added to config:

```
Requests:
      cpu:                            16200m
     <some-new-device-resource>:  1 # This is a new resource request applied to each pod.
      memory:                         6G
      nvidia.com/gpu:                 0
```

## CA Logs
I0312 21:38:06.717094       6 utils.go:208] Pod `<some-pod> `can't be scheduled on `<some-asg>` predicate failed: PodFitsResources predicate mismatch, reason: Insufficient `<some-new-device-resource>`

## Notes
ASGs with gpu instances are working fine as expected. But ASGs with cpu instances are **NOT** able to scale up from zero.

**Comment by Jeffwan** [comment]
Hi @xinxingliu90 We have the custom feature support for template node. Check here 
https://github.com/kubernetes/autoscaler/pull/1656/files and https://github.com/kubernetes/autoscaler/issues/1650

To use it, tag ASG node group like this
```
KEY = k8s.io/cluster-autoscaler/node-template/resources/devices.applecloud.io/fuse_v2
VALUE= 1 (number of resources on that node)
```

**Comment by Jeffwan** [comment]
BTW, does GPU pods need this resource? 

```
 devices.applecloud.io/fuse_v2:  1 # This is a new resource request applied to each pod.
```

**Comment by xinxingliu90** [comment]
@Jeffwan , yes, both GPU and CPU pods has the "fuse" resource in request.

**Comment by Jeffwan** [comment]
For you CPU pod, why do you have this resource request?

```
 nvidia.com/gpu:                 0
```

**Comment by xinxingliu90** [comment]
@Jeffwan That comes from our existing logic when creating a pod. CA used to work with this config.

**Comment by Jeffwan** [comment]
Please check above PR. Once it's merged, I will cherry-pick to 1.14. In case you want to test it now, I build one and you can try it `seedjeffwan/cluster-autoscaler:1.14.8-dev`

**Comment by xinxingliu90** [comment]
> Please check above PR. Once it's merged, I will cherry-pick to 1.14. In case you want to test it now, I build one and you can try it `seedjeffwan/cluster-autoscaler:1.14.8-dev`

CA works perfect with this image you provided. Thank you @Jeffwan !

**Comment by Jeffwan** [comment]
@xinxingliu90 

https://github.com/kubernetes/autoscaler/releases/tag/cluster-autoscaler-1.14.8
https://github.com/kubernetes/autoscaler/releases/tag/cluster-autoscaler-1.15.6

Try official releases.
